Pull request for bugfix/packaging

heucke opened the pull request:
Move .so link to -dev package
Fix libcurl3 dependency
Use libjs-jquery instead of bundled jquery.js
Add extended descriptions
Add missing python dependency
Fix headers package section
Add debian/copyright
Change virtual package depends to real package
Update Standards-Version
Add debian/source/format file
Sort Build-Depends

heucke committed:
Sort Build-Depends

* Simplify future modifications

Signed-off-by: Tyler Heucke <tyler@heucke.io>

heucke committed:
Add debian/source/format file

* Lintian: missing-debian-source-format

Signed-off-by: Tyler Heucke <tyler@heucke.io>

heucke committed:
Update Standards-Version

* Lintian: ancient-standards-version

Signed-off-by: Tyler Heucke <tyler@heucke.io>

heucke committed:
Change virtual package depends to real package

* Lintian: virtual-package-depends-without-real-package-depends

Signed-off-by: Tyler Heucke <tyler@heucke.io>

heucke committed:
Add debian/copyright

* Lintian: no-copyright-file

Signed-off-by: Tyler Heucke <tyler@heucke.io>

heucke committed:
Fix headers package section

* Lintian: wrong-section-according-to-package-name

Signed-off-by: Tyler Heucke <tyler@heucke.io>

heucke committed:
Add missing python dependency

* Lintian: python-script-but-no-python-dep

Signed-off-by: Tyler Heucke <tyler@heucke.io>

heucke committed:
Add extended descriptions

* Lintian: extended-description-is-empty

Signed-off-by: Tyler Heucke <tyler@heucke.io>

heucke commented:
Remaining issues:

```
E: opx-system-utils source: source-is-missing scripts/lib/python/scripts/fusioncharts.charts.js line length is 659 characters (>512)
E: opx-system-utils source: source-is-missing scripts/lib/python/scripts/fusioncharts.widgets.js line length is 520 characters (>512)
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/templates/basic_style.xhtml
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/images/opx.ico
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/templates/ma_chart_script.xhtml
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/templates/main.xhtml
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/templates/bulb_script.xhtml
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/plugins/__init__.py
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/templates/thermal_script.xhtml
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/images/dell.ico
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/plugins/temp.py
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/templates/chart_script.xhtml
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/templates/ms_chart_script.xhtml
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/templates/dial_script.xhtml
```

heucke committed:
Clean up autotools code

* Alphabetize automake script targets
* Use pkg-config for libcurl dependency
* Use standard include paths for opx-common-utils dep
* Remove unused code in configure.ac

Signed-off-by: Tyler Heucke <tyler@heucke.io>

heucke commented:
Pull-request updated, HEAD is now 76fe70257363a422d2f01076c28d92e91cda9f30

rakeshdatta commented on debian/libopx-system-utils1.install:
```diff
@@ -1,2 +1 @@
-usr/lib/*/*.so*
-usr/lib/*/*.so
+usr/lib/*/*.so.*
```
This is not going to include the so files with no version suffix.


```
rdatta@xenlogin-eqx-01:/neteng/rdatta/system-utils/pool/stretch-amd64/opx-system-utils$ ls -l libopx-system-utils1_0.1.0_amd64.deb /usr/lib/*.so
-rw-r--r-- 1 rdatta ncore  3262 Nov  8 12:06 libopx-system-utils1_0.1.0_amd64.deb
-rw-r--r-- 1 root   root  39720 Apr 14  2016 /usr/lib/attica_kde.so
lrwxrwxrwx 1 root   root     10 Jan 22  2016 /usr/lib/libau.so -> libau.so.2
lrwxrwxrwx 1 root   root     25 Feb 10  2016 /usr/lib/libganglia.so -> libganglia-3.6.0.so.0.0.0
-rw-r--r-- 1 root   root  27080 Apr 14  2016 /usr/lib/libknotifyplugin.so
lrwxrwxrwx 1 root   root     27 Feb 17  2016 /usr/lib/libxmlsec1-gcrypt.so -> libxmlsec1-gcrypt.so.1.2.20
lrwxrwxrwx 1 root   root     27 Feb 17  2016 /usr/lib/libxmlsec1-gnutls.so -> libxmlsec1-gnutls.so.1.2.20
lrwxrwxrwx 1 root   root     24 Feb 17  2016 /usr/lib/libxmlsec1-nss.so -> libxmlsec1-nss.so.1.2.20
lrwxrwxrwx 1 root   root     28 Feb 17  2016 /usr/lib/libxmlsec1-openssl.so -> libxmlsec1-openssl.so.1.2.20
lrwxrwxrwx 1 root   root     20 Feb 17  2016 /usr/lib/libxmlsec1.so -> libxmlsec1.so.1.2.20

```

heucke replied:
Unversioned libraries come from the dev package.

https://github.com/open-switch/opx-system-utils/pull/2/files#diff-0ea4544c9cea3b292b01410766187affL1

See: https://lintian.debian.org/tags/non-dev-pkg-with-shlib-symlink.html

heucke replied:
Also see this: https://lists.debian.org/debian-mentors/2006/03/msg00356.html

rakeshdatta commented on scripts/lib/python/plugins/analytics.py:
```diff
@@ -25,7 +25,7 @@
 	 viewBox="0 0 505 505" style="enable-background:new 0 0 505 505;" xml:space="preserve">
```
This file have changed.. You need to rebase your branch off my branch

heucke replied:
Yes, I need to some work on this PR before it merges. I should be able to get to it this week.

rakeshdatta reviewed (changes requested):
Files and their contents have changed off my feature branch. Please rebase your bugfix branch.

heucke committed:
One dependency per line

Signed-off-by: Tyler Heucke <tyler.heucke@dell.com>

heucke committed:
Fix libcurl dependency name

Signed-off-by: Tyler Heucke <tyler.heucke@dell.com>

heucke committed:
Move .so link to -dev package

* Lintian: non-dev-pkg-with-shlib-symlink

Signed-off-by: Tyler Heucke <tyler.heucke@dell.com>

heucke commented:
Pull-request updated, HEAD is now 214e3216fe715300eb553a7b023cba4bcd5b1be1

heucke commented:
Rebased and fixed up. Remaining issues:
```bash
Now running lintian...
E: opx-system-utils source: source-is-missing scripts/lib/python/scripts/Chart.min.js
E: opx-system-utils source: source-is-missing scripts/lib/python/scripts/gauge.min.js
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/templates/basic_style.xhtml
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/templates/main.xhtml
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/templates/alarm.xhtml
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/templates/dial.xhtml
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/templates/temp.xhtml
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/images/dell.ico
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/images/opx.ico
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/plugins/__init__.py
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/plugins/temp.py
W: opx-system-utils: executable-not-elf-or-script usr/lib/opx/sys-utils/templates/chart.xhtml
Finished running lintian.
```

heucke commented:
You'll definitely want to squash this PR when merging, I just left the commits separately so you can more easily review the changes.

heucke commented:
File lists:
```
$ dpkg-deb -c libopx-system-utils-dev_0.1.0_amd64.deb
drwxr-xr-x root/root         0 2018-10-23 15:10 ./
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/include/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/include/opx/
-rw-r--r-- root/root      3433 2018-10-23 15:10 ./usr/include/opx/sys_utils_file_transfer.h
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/lib/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/lib/x86_64-linux-gnu/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/share/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/share/doc/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/share/doc/libopx-system-utils-dev/
-rw-r--r-- root/root       153 2018-10-23 15:10 ./usr/share/doc/libopx-system-utils-dev/changelog.gz
-rw-r--r-- root/root       947 2018-10-23 15:10 ./usr/share/doc/libopx-system-utils-dev/copyright
lrwxrwxrwx root/root         0 2018-10-23 15:10 ./usr/lib/x86_64-linux-gnu/libopx_system_utils.so -> libopx_system_utils.so.1.0.1
```

```
$ dpkg-deb -c libopx-system-utils1_0.1.0_amd64.deb
drwxr-xr-x root/root         0 2018-10-23 15:10 ./
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/lib/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/lib/x86_64-linux-gnu/
-rw-r--r-- root/root     10080 2018-10-23 15:10 ./usr/lib/x86_64-linux-gnu/libopx_system_utils.so.1.0.1
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/share/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/share/doc/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/share/doc/libopx-system-utils1/
-rw-r--r-- root/root       153 2018-10-23 15:10 ./usr/share/doc/libopx-system-utils1/changelog.gz
-rw-r--r-- root/root       947 2018-10-23 15:10 ./usr/share/doc/libopx-system-utils1/copyright
lrwxrwxrwx root/root         0 2018-10-23 15:10 ./usr/lib/x86_64-linux-gnu/libopx_system_utils.so.1 -> libopx_system_utils.so.1.0.1
```

```
$ dpkg-deb -c opx-system-utils_0.1.0_amd64.deb
drwxr-xr-x root/root         0 2018-10-23 15:10 ./
drwxr-xr-x root/root         0 2018-10-23 15:10 ./lib/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./lib/systemd/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./lib/systemd/system/
-rw-r--r-- root/root       395 2018-10-23 15:10 ./lib/systemd/system/opx-system-utils-db.service
-rw-r--r-- root/root       387 2018-10-23 15:10 ./lib/systemd/system/opx-system-utils.service
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/lib/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/lib/opx/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/lib/opx/sys-utils/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/lib/opx/sys-utils/images/
-rw-r--r-- root/root     10419 2018-10-23 15:10 ./usr/lib/opx/sys-utils/images/SNMP_blue.png
-rwxr-xr-x root/root    269342 2018-10-23 15:10 ./usr/lib/opx/sys-utils/images/dell.ico
-rwxr-xr-x root/root      9243 2018-10-23 15:10 ./usr/lib/opx/sys-utils/images/opx.ico
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/lib/opx/sys-utils/plugins/
-rwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/lib/opx/sys-utils/plugins/__init__.py
-rwxr-xr-x root/root     18755 2018-10-23 15:10 ./usr/lib/opx/sys-utils/plugins/analytics.py
-rwxr-xr-x root/root      8552 2018-10-23 15:10 ./usr/lib/opx/sys-utils/plugins/cpu.py
-rwxr-xr-x root/root      7748 2018-10-23 15:10 ./usr/lib/opx/sys-utils/plugins/disk.py
-rwxr-xr-x root/root      9793 2018-10-23 15:10 ./usr/lib/opx/sys-utils/plugins/fan.py
-rwxr-xr-x root/root      7990 2018-10-23 15:10 ./usr/lib/opx/sys-utils/plugins/mem.py
-rwxr-xr-x root/root      9674 2018-10-23 15:10 ./usr/lib/opx/sys-utils/plugins/power.py
-rwxr-xr-x root/root      8402 2018-10-23 15:10 ./usr/lib/opx/sys-utils/plugins/swap.py
-rwxr-xr-x root/root     11024 2018-10-23 15:10 ./usr/lib/opx/sys-utils/plugins/temp.py
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/lib/opx/sys-utils/scripts/
-rw-r--r-- root/root    345182 2018-10-23 15:10 ./usr/lib/opx/sys-utils/scripts/Chart.min.js
-rw-r--r-- root/root     70411 2018-10-23 15:10 ./usr/lib/opx/sys-utils/scripts/gauge.min.js
-rwxr-xr-x root/root     70964 2018-10-23 15:10 ./usr/lib/opx/sys-utils/sys_status.py
-rwxr-xr-x root/root      4126 2018-10-23 15:10 ./usr/lib/opx/sys-utils/system_status_db.py
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/lib/opx/sys-utils/templates/
-rwxr-xr-x root/root      1570 2018-10-23 15:10 ./usr/lib/opx/sys-utils/templates/alarm.xhtml
-rwxr-xr-x root/root       780 2018-10-23 15:10 ./usr/lib/opx/sys-utils/templates/basic_style.xhtml
-rwxr-xr-x root/root      2744 2018-10-23 15:10 ./usr/lib/opx/sys-utils/templates/chart.xhtml
-rwxr-xr-x root/root      3300 2018-10-23 15:10 ./usr/lib/opx/sys-utils/templates/dial.xhtml
-rwxr-xr-x root/root      6901 2018-10-23 15:10 ./usr/lib/opx/sys-utils/templates/main.xhtml
-rwxr-xr-x root/root      3016 2018-10-23 15:10 ./usr/lib/opx/sys-utils/templates/temp.xhtml
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/share/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/share/doc/
drwxr-xr-x root/root         0 2018-10-23 15:10 ./usr/share/doc/opx-system-utils/
-rw-r--r-- root/root       153 2018-10-23 15:10 ./usr/share/doc/opx-system-utils/changelog.gz
-rw-r--r-- root/root       947 2018-10-23 15:10 ./usr/share/doc/opx-system-utils/copyright
```
